File: node/grabber.js
var request = require('request');
var cheerio = require('cheerio');
var fs = require('fs');
var result = {};
var ghin_nums = ['2064561','2259440','0878514'];

ghin_nums.forEach(function(ghin_num) {
  request('http://m.ghin.com/HLR.aspx?ghinno=' + ghin_num, function(error, response, body) {
    console.log("Error: ", error);
    console.log("StatusCode: ", response && response.statusCode);
    // console.log("Body: ", body);
    // fs.writeFile('page.html', body, 'utf8');
    $ = cheerio.load(body);
    var h = $("#ctl00_cphContent_lblHI").text();
    h = parseFloat(h);
    var n = $("#ctl00_cphContent_lblName").text();
    var c = $("#ctl00_cphContent_lblClub").text();
    result[ghin_num] = {name: n, handicap: h, club: c};
    console.log(result);
  });
});

File: python/grabber.py
from pyquery import PyQuery as pq

base_url = "http://m.ghin.com/HLR.aspx?ghinno="
ghin_nums = ['2064561','2259440','0878514']
result = {}

for ghin in ghin_nums:
    d = pq(url = base_url + ghin)
    h = d("#ctl00_cphContent_lblHI").text();
    n = d("#ctl00_cphContent_lblName").text();
    c = d("#ctl00_cphContent_lblClub").text();
    result[ghin] = {
        'name': n,
        'handicap': h,
        'club': c
    }

for k in result.keys():
    print "GHIN: %s\n%r\n" % (k, result[k])
